elastic/beats
# Allow the Agent to retrieve configuration from the Fleet API.

Correctly Fetch a configuration from the Fleet API and send everything to the operator.
This PR doesn't include errors reportings.

Ref: #14951

depends on: #15089


--- 

How to test?

- Use the dev-tools/cmd/fakewebapi
- sh push.sh to push a configuration to the API.
- mage build:dev
- ./agent enroll http://localhost:8080 abc123
- Adjust the path in agent.yml see #15167 for followup.
- BEATS_VERSION=7.5.0 agent run
